rosbag2_storage/src/rosbag2_storage/metadata_io.cpp

```diff
@@ -73,14 +96,42 @@ struct convert<rosbag2_storage::TopicInformation>
     return node;
   }
 
-  static bool decode(const Node & node, rosbag2_storage::TopicInformation & metadata)
+  static bool decode(const Node & node, rosbag2_storage::TopicInformation & metadata, int version)
   {
-    metadata.topic_metadata = node["topic_metadata"].as<rosbag2_storage::TopicMetadata>();
+    metadata.topic_metadata = decode_for_version<rosbag2_storage::TopicMetadata>(
+      node["topic_metadata"], version);
     metadata.message_count = node["message_count"].as<uint64_t>();
     return true;
   }
 };
 
+template<>
+struct convert<std::vector<rosbag2_storage::TopicInformation>>
+{
+  static Node encode(const std::vector<rosbag2_storage::TopicInformation> & rhs)
+  {
+    Node node(NodeType::Sequence);
+    for (auto value : rhs) {
+      node.push_back(value);
+    }
+    return node;
+  }
+
+  static bool decode(
+    const Node & node, std::vector<rosbag2_storage::TopicInformation> & rhs, int version)
+  {
+    if (!node.IsSequence()) {
+      return false;
+    }
+
+    rhs.clear();
+    for (const auto & value : node) {
+      rhs.push_back(decode_for_version<rosbag2_storage::TopicInformation>(value, version));
+    }
```

Comment:
same comment as above. Can we initialize `rhs` in place instead of iterating over node?

Reply:
unfortunately it does not seem possible with `yaml-cpp`